lib/setup/setupInstall.js

```diff
@@ -538,37 +538,34 @@ function Install(options) {
 
     // this command is executed always on THIS host
     function checkDependencies(adapter, deps, globalDeps, _options, callback) {
-        if ((!deps || !deps.length) && (!globalDeps || !globalDeps.length)) {
+        if (!deps && !globalDeps) {
             return callback && callback(adapter);
         }
 
-        // combine both dependencies
-        deps = (deps || []).concat(globalDeps || []);
+        deps = tools.parseDependencies(deps);
+        globalDeps = tools.parseDependencies(globalDeps);
+
+        // combine both dependencies for now
+        Object.assign(deps, globalDeps);
 
         let cnt = 0;
         // Get all installed adapters
         objects.getObjectView('system', 'instance', {}, null, (err, objs) => {
             err && console.error(err);
 
             if (objs && objs.rows && objs.rows.length) {
-                for (let i = 0; i < deps.length; i++) {
-                    let dName;
+                for (const dName in deps) {
                     let version = null;
                     let isFound = false;
 
-                    if (typeof deps[i] === 'object') {
-                        dName = Object.keys(deps[i])[0];
-                        version = deps[i][dName];
-                    } else {
-                        dName = deps[i];
-                    }
+                    version = deps[dName];
 
                     if (dName === 'js-controller') {
                         // Check only version
                         if (version) {
```

Comment:
We need to adjust that. Now it is "*" for any version (compared to null before)

Reply:
Not really a problem, because `*` means all versions for semver, but yes some null checks are unnecessary now. Thanks